beacon-chain/blockchain/process_block.go

```diff
@@ -321,30 +321,31 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []block.SignedBeaconBlo
 		}
 		jCheckpoints[i] = preState.CurrentJustifiedCheckpoint()
 		fCheckpoints[i] = preState.FinalizedCheckpoint()
-		sigSet.Join(set)
-	}
-	verify, err := sigSet.Verify()
-	if err != nil {
-		return nil, nil, err
-	}
-	if !verify {
-		return nil, nil, errors.New("batch block signature verification failed")
-	}
 
-	// blocks have been verified, add them to forkchoice and call the engine
-	for i, b := range blks {
 		preStateVersion, preStateHeader, err := getStateVersionAndPayload(preState)
 		if err != nil {
 			return nil, nil, err
 		}
-
 		s.saveInitSyncBlock(blockRoots[i], b)
 		if err := s.insertBlockToForkChoiceStore(ctx, b.Block(), blockRoots[i], fCheckpoints[i], jCheckpoints[i]); err != nil {
 			return nil, nil, err
 		}
 		if err := s.notifyNewPayload(ctx, preStateVersion, preStateHeader, preState, b, blockRoots[i]); err != nil {
 			return nil, nil, err
 		}
+
+		sigSet.Join(set)
+	}
+	verify, err := sigSet.Verify()
```

Comment:
What happens if the signature does not verify, don't we end up with a bad block in forkchoice anyway?